examples/cluster/plot_kmeans_silhouette_analysis.py

```diff
@@ -0,0 +1,123 @@
+"""
+===============================================================================
+Silhouette analysis for sample data clustered using KMeans clustering algorithm
+===============================================================================
+
+Silhouette analysis can be used to study the separation distance between the
+resulting clusters. The silhouette plot displays a measure of how close each
```

Comment:
I thought grid search of all the clustering metrics could be compared and presented for #4040 

For this example could I add a plot of `silhouette scores` vs `n_clusters` for a range of 2 to 8, instead?